pkg/controller/common/volume/downward_api.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package volume
+
+import (
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/volume"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var downwardApiVolume = corev1.Volume{
+	Name: volume.DownwardApiVolumeName,
+	VolumeSource: corev1.VolumeSource{
+		DownwardAPI: &corev1.DownwardAPIVolumeSource{
+			Items: []corev1.DownwardAPIVolumeFile{
+				{
+					Path: volume.LabelsFile,
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.labels",
+					},
+				},
+			},
+		},
+	},
+}
+
+var downwardApiVolumeMount = corev1.VolumeMount{
+	Name:      volume.DownwardApiVolumeName,
+	MountPath: volume.DownwardApiMountPath,
+	ReadOnly:  true,
+}
```

Comment:
Nit: I feel like these don't need to be global variables (even though they're only package-private).  Maybe they can be directly returned from the method calls of the `DownwardApi` object?

Reply:
The idea here is to avoid to recreate what must be a constant everytime a Pod spec is created. I understand that it may sound like a very tiny optimization though 😐

Reply:
:+1:  I don't feel strongly about it. Happy to leave it as is.